### PR TITLE
Adds the HF Blade, removes the unused one from the code

### PR DESCRIPTION
**Pull request by karmaisblackandbluepilled**
it dismembers and that's it
I also removed the old adminspawn only one
## Changelog
:cl:
add: think something up funny for me I am lazy
/:cl:

### CODE REVIEW

File: code/game/objects/items/holy_weapons.dm [hunk 1]
```diff
@@ -407,7 +407,7 @@
 	AddComponent(/datum/component/butchering, 70, 110) //the harvest gives a high bonus chance
 
 /obj/item/nullrod/scythe/vibro
-	name = "high frequency blade"
+	name = "high fronquency bladd"
```
Comment (EdgeLordExe) [inline code/game/objects/items/holy_weapons.dm:410]: What

File: code/game/objects/items/melee/hf.dm [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+/obj/item/melee/hfblade/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/melee/hfblade/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(attack_type == PROJECTILE_ATTACK)
+		final_block_chance = 0
+	if(attack_type == MELEE_ATTACK) // It really is not a good idea to get close to this thing.
+		final_block_chance = 100
```
Comment (EdgeLordExe) [inline code/game/objects/items/melee/hf.dm:40]: Not sure about this. Id accept 75% but 100 is a bit much

File: code/game/objects/items/melee/hf.dm [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+				if(BP.dismemberable)
+					parts += BP
+	if(parts)
+		if(!parts.len)
+			return FALSE
+		var/mob/living/carbon/C = target
+		var/obj/item/bodypart/BP = pick(parts)
+		BP.dismember()
+		playsound(C,pick('sound/misc/desecration-01.ogg','sound/misc/desecration-02.ogg','sound/misc/desecration-01.ogg') ,50, 1, -1)
+		C.adjustBruteLoss(-15) //dismembering a limb deals 15 brute, this heals you instantly after that
```
Comment (EdgeLordExe) [inline code/game/objects/items/melee/hf.dm:72]: This may have been changed with wounds

File: code/modules/uplink/uplink_items.dm [hunk 4]
```diff
@@ -2126,3 +2126,11 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	item = /obj/item/melee/execution_sword
 	cost = 1 //it sucks balls for anything but memeing
 	surplus = 30 //Theres a good chance this will end up in surplus crates, so its a great way to add a little spice to any meme round.
+
+/datum/uplink_item/dangerous/vibroblade
+	name = "High Frequency Blade"
+	desc = "An electric katana that weakens the molecular bonds of whatever it touches. Perfect for slicing off the limbs of your coworkers. \
+	Avoid using a multitool on it."
+	item = /obj/item/storage/belt/hfblade
+	cost = 9
```
Comment (EdgeLordExe) [inline code/modules/uplink/uplink_items.dm:2135]: Way too low. Make it at least 15 tc

**Review by EdgeLordExe — CHANGES_REQUESTED** [review]
Balancing concerns

File: code/game/objects/items/holy_weapons.dm (hunk 1 above)
Comment (Shadowfgt) [reply code/game/objects/items/holy_weapons.dm:410]: They probably changed the name so it would not be confused with the real thing and get a chap lynched

Comment (karmaisblackandbluepilled) [reply code/game/objects/items/holy_weapons.dm:410]: spot on

File: code/modules/uplink/uplink_items.dm (hunk 4 above)
Comment (karmaisblackandbluepilled) [reply code/modules/uplink/uplink_items.dm:2135]: ok

File: code/game/objects/items/melee/hf.dm (hunk 3 above)
Comment (karmaisblackandbluepilled) [reply code/game/objects/items/melee/hf.dm:72]: yeah hold on lemme test if that's true
somehow
uh

File: code/game/objects/items/melee/hf.dm (hunk 2 above)
Comment (karmaisblackandbluepilled) [reply code/game/objects/items/melee/hf.dm:40]: sleeping carp gives you 100%

File: code/game/objects/items/melee/hf.dm (hunk 3 above)
Comment (karmaisblackandbluepilled) [reply code/game/objects/items/melee/hf.dm:72]: ok I understand how to test now

**Comment by AcapitalA** [comment]
If you're gonna do this, at least do something with the old HF blade, such as removing it.

**Comment by karmaisblackandbluepilled** [comment]
the old hf blade is a nullrod, why would I remove it

**Comment by karmaisblackandbluepilled** [comment]
oh I see what you're talking about
why would I remove that one, it's an actual weapon that deals damage instead of dismembering

**Comment by AcapitalA** [comment]
It's just weird that there's 2 weapons with the same name and identical icons are in the game, it'll also be confusing for admins and players alike considering the block chances of both.

**Comment by cuckooman** [comment]
someone did alternate sprites for Sam's sword - that could be used for one of them
https://media.discordapp.net/attachments/878737577748856832/957737593561284679/unknown.png

**Comment by cuckooman** [comment]
for more differentiation

**Comment by AcapitalA** [comment]
> someone did alternate sprites for Sam's sword - that could be used for one of them https://media.discordapp.net/attachments/878737577748856832/957737593561284679/unknown.png

Agreed, best solution right here. Hell, maybe even special interactions for both.

**Comment by karmaisblackandbluepilled** [comment]
are you both retarded

**Comment by cuckooman** [comment]
> are you both retarded

quote possibly

**Comment by AcapitalA** [comment]
what

File: code/game/objects/items/melee/hf.dm [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+/obj/item/melee/hfblade/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(attack_type == PROJECTILE_ATTACK)
+		final_block_chance = 0
+	if(attack_type == MELEE_ATTACK) // It really is not a good idea to get close to this thing.
+		final_block_chance = 100
+		var/hideandseeklogic = rand(1,5) // YES, ADD MORE RNG TO COMBAT.
+		if(ishuman(hitby.loc))
+			var/mob/living/carbon/human/attacker = hitby.loc
+			if(hideandseeklogic == 1 || HAS_TRAIT(attacker, TRAIT_CLUMSY))
```
Comment (TheChosenEvilOne) [inline code/game/objects/items/melee/hf.dm:44]:
```suggestion
		if(ishuman(hitby.loc))
			var/mob/living/carbon/human/attacker = hitby.loc
			if(prob(20) || HAS_TRAIT(attacker, TRAIT_CLUMSY))
```

File: code/game/objects/items/melee/hf.dm [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+	if(!proximity_flag && rules_of_nature)
+		return
+	var/list/parts = list()
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		for(var/X in C.bodyparts)
+			var/obj/item/bodypart/BP = X
+			if(BP.body_part != HEAD && BP.body_part != CHEST)
+				if(BP.dismemberable)
+					parts += BP
```
Comment (TheChosenEvilOne) [inline code/game/objects/items/melee/hf.dm:63]:
```suggestion
		for(var/X in C.bodyparts)
			var/obj/item/bodypart/BP = X
			if(!BP.dismemberable)
				continue
			if(!(BP.body_part & (HEAD | CHEST)))
				parts += BP
```

**Comment by JinGiant** [comment]
Give me maintainer so I can add the do not merge tag
This pr fucking sucks

**Comment by karmaisblackandbluepilled** [comment]
ok so I think I made it have different block changes based on if you're wielding it or not.
33 melee 80 ranged if wielded
80 melee 20 ranged if not

**Pull request merged**